firmware/hw_layer/AdcConfiguration.h

```diff
@@ -22,8 +22,9 @@ class AdcDevice {
 	int conversionCount;
 	int errorsCount;
 	int getAdcValueByIndex(int internalIndex) const;
+	void invalidateSamplesCache();
 
-	adcsample_t samples[ADC_MAX_CHANNELS_COUNT * MAX_ADC_GRP_BUF_DEPTH];
+	__ALIGNED(32) adcsample_t samples[ADC_MAX_CHANNELS_COUNT * MAX_ADC_GRP_BUF_DEPTH];
```

Comment:
I believe that this code is vague. Quality code would self-document WHY do we suddenly have to do an unusual thing. I believe that either a human-readable comment referring to F7 caches or MAGIC_F7_DMA_FLASH mentioning is needed here